src/app/Library/CrudPanel/Traits/Query.php

```diff
@@ -148,4 +150,105 @@ public function orderByWithPrefix($column_name, $column_direction = 'ASC')
 
         return $this->query->orderBy($column_name, $column_direction);
     }
+
+    /**
+     * return the nested query columns.
+     *
+     * @param  \Illuminate\Database\Eloquent\Builder  $query
+     * @return array
+     */
+    public function getNestedQueryColumns($query)
+    {
+        return $this->getQueryColumnsFromWheres($query, true);
+    }
+
+    /**
+     * we want to select the minimum possible columns respecting the clauses in the query, so that the count is accurate.
+     * for that to happen we will traverse the query `wheres` (Basic, Exists, Nested or Column) to get the correct
+     * column that we need to select in the main model for that "sub query" to work.
+     *
+     * For example a base query of: `SELECT * FROM table WHERE (someColumn, smtValue)`, we would return only the `someColumn`
+     * with the objective of replacing the `*` for the specific columns needed, avoiding the selection of
+     * columns that would not have impact in the counting process.
```

Comment:
Freaking EXCELLENT explanation here Pedro. 👏👏👏 Just one nitpick:

```suggestion
     * We want to select the minimum possible columns respecting the clauses in the query, so that the count is accurate.
     * for that to happen we will traverse the query `wheres` (Basic, Exists, Nested or Column) to get the correct
     * column that we need to select in the main model for that "sub query" to work.
     *
     * For example a base query of: `SELECT * FROM table WHERE (someColumn, smtValue)`, we would return only the `someColumn`
     * with the objective of replacing the `*` for the specific columns needed, avoiding the selection of
     * columns that would not have impact in the counting process.
```